File: CollegeDatabase/Persons/StudentDecorator.cs
namespace CollegeDatabase.Persons
{
    /// <summary>
    /// Dekorator
    /// </summary>
    public class StudentDecorator
    {
        Student Student { get; set; }

        public string FirstName
        {
            get
            {
                return this.Student.FirstName + " (udekorowane imie Studenta)";
            }
        }

        public StudentDecorator(Student student)
        {
            this.Student = student;
        }

        /// <summary>
        /// Udekoruj metode - dodaj coś do podstawowej metody z obiektu przekazanego w konstruktorze.
        /// </summary>
        public override string ToString()
        {
            return this.Student + " - udekorowany";
        }
    }
}

File: CollegeDatabasePython/StudentDecorator.py
from Student import Student

class StudentDecorator(object):

    def __init__(self, student):
        self.Student = student

    def FirstName(self):
        return self.Student.FirstName + " (udekorowane imie Studenta)"

    def ToString(self):
        return self.Student.ToString() + " - udekorowany";